tests/legacy-cli/e2e/tests/experimental/ivy-lazy-load.ts

```diff
@@ -5,23 +5,88 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
-import { readdirSync } from 'fs';
-import { prependToFile, replaceInFile } from '../../utils/fs';
+import { prependToFile, replaceInFile, writeFile } from '../../utils/fs';
 import { ng } from '../../utils/process';
-import { createProject } from '../../utils/project';
+import { createProject, updateJsonFile } from '../../utils/project';
+import { expectToFail } from '../../utils/utils';
 
-export default async function() {
-  await createProject('ivy-project', '--enable-ivy');
+export async function ivyLazyLoadSetup(projectName: string) {
+  // Make Ivy project.
+  await createProject(projectName, '--enable-ivy');
 
+  // Add lazy route.
   await ng('generate', 'module', 'lazy', '--routing');
   await prependToFile('src/app/app.module.ts', `import { RouterModule } from '@angular/router';`);
   await replaceInFile('src/app/app.module.ts', 'imports: [', `imports: [
-    RouterModule.forRoot([{ path: "lazy", loadChildren: "src/app/lazy/lazy.module#LazyModule" }]),
+    RouterModule.forRoot([{ path: 'lazy', loadChildren: LOAD_CHILDREN_MARKER }]),
   `);
-  await ng('build', '--named-chunks', '--aot');
-  const distFiles = await readdirSync('dist/ivy-project');
-  if (!distFiles.includes('lazy-lazy-module.js')
-    || distFiles.includes('lazy-lazy-module-ngfactory.js')) {
-    throw new Error('The lazy module chunk should not be a ngfactory.');
-  }
+  await ng('generate', 'component', 'lazy/lazy-comp');
+  await replaceInFile('src/app/lazy/lazy-routing.module.ts', 'const routes: Routes = [];', `
+    import { LazyCompComponent } from './lazy-comp/lazy-comp.component';
+    const routes: Routes = [{ path: '', component: LazyCompComponent }];
+  `);
+  await writeFile('src/app/app.component.html', '<router-outlet></router-outlet>');
+
+  // Allow `import()` calls.
+  await replaceInFile('tsconfig.json', `"module": "es2015"`, `"module": "esnext"`);
```

Comment:
If this is required, should we scafold project with this option?

Reply:
For ivy apps, yes.  Otherwise dynamic imports aren't supported by TypeScript.

Reply:
We support dynamic imports overall really, I think since 6 or something. Maybe we should just add that to the default for all projects.

Reply:
It also enables `import.meta` which webpack doesn't currently like.  But the user would need to actively use it for that to be a problem and its not quite common yet.

Reply:
Added a separate commit to use `"module": "esnext"` as the default. It's after the others so we can remove it without much bother, or move it to a separate PR. Let me know what you prefer.

Reply:
Can you add a brief description of what effects the change has in the commit body?  Should help with any questions after release